### PR TITLE
Replace wherever possible '.reset(new' with '= make_unique'

### PR DESCRIPTION
**Pull request by lucafedeli88**
Following https://github.com/ECP-WarpX/WarpX/pull/1398, this PR replaces code like `uptr.reset ( new Type(vars))` with `uptr = make_unique<Type>(vars)` everywhere in the code, except in very few cases where a custom deleter is used (e.g. for most plasma injectors and for shared pointers to arrays in `WarpXOpenPMD.cpp`)

### CODE REVIEW

File: Source/Diagnostics/WarpXOpenPMD.cpp [hunk 1]
```diff
@@ -470,10 +470,9 @@ WarpXOpenPMDPlot::DumpToFile (WarpXParticleContainer* pc,
            }
 #else
            for (auto currDim = 0; currDim < AMREX_SPACEDIM; currDim++) {
-                std::shared_ptr< amrex::ParticleReal > curr(
-                    new amrex::ParticleReal[numParticleOnTile],
-                    [](amrex::ParticleReal const *p){ delete[] p; }
-                );
+                std::shared_ptr<amrex::ParticleReal> curr =
+                    std::make_unique<amrex::ParticleReal[]>(numParticleOnTile);
+
```
Comment (lucafedeli88) [inline Source/Diagnostics/WarpXOpenPMD.cpp:475]: This solution leaves me a bit unhappy. But we can't make it cleaner until C++20. If it is too misleading we might go back to the original version.

File: Source/Diagnostics/WarpXOpenPMD.cpp [hunk 2]
```diff
@@ -435,10 +436,11 @@ WarpXOpenPMDPlot::DumpToFile (WarpXParticleContainer* pc,
...
 #if defined(WARPX_DIM_RZ)
            {
-              std::shared_ptr<amrex::ParticleReal> z(
-                      new amrex::ParticleReal[numParticleOnTile],
-                      [](amrex::ParticleReal const *p) { delete[] p; }
-              );
+               //for std::make_shared<T[]> we should wait C++20
+               //in the meanwhile, a unique_ptr can be converted in a shared_ptr
+               std::shared_ptr<amrex::ParticleReal> z =
+                std::make_unique<amrex::ParticleReal[]>(numParticleOnTile);
```
Comment (lucafedeli88) [inline Source/Diagnostics/WarpXOpenPMD.cpp:442]: This solution leaves me a bit unhappy. But we can't make it cleaner until C++20. If it is too misleading we might go back to the original version.

File: Source/Diagnostics/WarpXOpenPMD.cpp [hunk 3]
```diff
@@ -435,10 +436,11 @@ WarpXOpenPMDPlot::DumpToFile (WarpXParticleContainer* pc,
...
            {
-              std::shared_ptr<amrex::ParticleReal> z(
-                      new amrex::ParticleReal[numParticleOnTile],
-                      [](amrex::ParticleReal const *p) { delete[] p; }
-              );
+               //for std::make_shared<T[]> we should wait C++20
+               //in the meanwhile, a unique_ptr can be converted in a shared_ptr
+               std::shared_ptr<amrex::ParticleReal[]> z =
+                std::make_unique<amrex::ParticleReal[]>(numParticleOnTile);
+
```
Comment (lucafedeli88) [inline Source/Diagnostics/WarpXOpenPMD.cpp:443]: This solution leaves me a bit unhappy. But we can't make it cleaner until C++20. If it is too misleading we might go back to the original version.

**Review by ax3l — APPROVED** [review]
Thank you! :sparkles:

**Pull request merged**